kbs/src/attestation/backend.rs

```diff
@@ -324,23 +349,41 @@ impl AttestationService {
             }
         }
 
-        let token = self
-            .inner
-            .verify(evidence_to_verify)
-            .await
-            .context("verify TEE evidence failed")?;
-
         let mut session = self
             .session_map
             .sessions
             .get_async(session_id)
             .await
-            .ok_or(anyhow!("session not found"))?;
+            .ok_or(anyhow!("session not found"))
+            .inspect_err(|_| ATTESTATION_ERRORS.inc())?;
         let session = session.get_mut();
 
+        let tee_type_label = {
+            // it seems impossible to prevent serde from putting double-quotes
+            // around the tee name, get rid of them in a subsequent step
+            let quoted = serde_json::to_string(&session.request().tee)?;
+            quoted[1..quoted.len() - 1].to_owned()
+        };
+
+        let token = self
+            .inner
+            .verify(evidence_to_verify)
+            .await
+            .inspect_err(|_| {
+                ATTESTATION_FAILURES
+                    .with_label_values(&[&tee_type_label])
```

Comment:
At this point we aren't guaranteed to have a valid tee type so there is a chance of dos.

Reply:
actually maybe it's fine since we've deserialized the Tee enum